cmd/zfs/zfs_main.c

```diff
@@ -6769,18 +6770,29 @@ get_one_dataset(zfs_handle_t *zhp, void *data)
 		zfs_close(zhp);
 		return (0);
 	}
+
+	/*
+	 * Skip any dataset whos name doesn't have a prefix matching ga_dataset.
+	 */
+	if (state->ga_dataset != NULL && strncmp(zfs_get_name(zhp), 
```

Comment:
Not sure what `zfs_get_name(zhp)` returns, but do you need an additional check that `strlen(state->ga_dataset) < strlen(zfs_get_name(zhp))`?

Reply:
I am guessing, but the use of `strlen()` in there is probably looking for datasets beneath a mountpoint. So if it starts with `ga_dataset` go and process it. `strncmp("/mount/point/lower", "/mount/point", 12)`

Reply:
`zfs_get_name(zhp)`: I understand it to return the datasets's path `zpool/path/to/dataset` (AI)
`strncmp()`: _compares two strings character by character_
`strlen()` is used limit the compare to the length of `state->ga_dataset` (basically a prefix match)

The logic being that if you want to mount everything under `zpool/path` recursively
 - check the name of the current dataset
 - if the datasets name starts with `zpool/path` then continue (with the function) 
 - else `return (0)` (discard this iteration)